systemd-journal: fix --with-systemd-journal=optional mode

HofiOne opened the pull request:
Further modifications in the lookup of sd_id128_XXX symbols (that can be presented in separate modules on some OSes e.g. Centos)

Signed-off-by: Hofi <hofione@gmail.com>

bazsi commented on modules/systemd-journal/journald-subsystem.c:
```diff
@@ -57,20 +57,25 @@ int (*sd_id128_get_boot)(sd_id128_t *ret);
+const char *journald_ext_libraries[] = {ID128_LIBRARY_NAME, JOURNAL_LIBRARY_NAME, SYSTEMD_LIBRARY_NAME, NULL};
 
 static GModule *journald_module;
+static GModule *journald_ext_module;
```
we could probably move these to local variables in load_journald_subsystem() and then pass them along
using arguments to _load_journald_subsystem()

HofiOne replied:
yeah, it came to my mind as well, the only reason I left it this way is that once we might want to add a really proper cleanup then we might want to call unload_journald_subsystem() and that would not be that nice to pass these to that function too

bazsi reviewed (approved):
I am fine with this as is, I added a small improvement that would have made reviewing this easier but that's optional too.

With that change, this unit would be drastically more readable than it has been a month ago :)

Thanks

HofiOne commented:
@bazsi Sorry, I've made a typo in libsystemd-id128.so.0, corrected, and added some logging of the lookup process and possible errors. Hope it's fine now.

HofiOne commented:
Centos tests are passed, seems we are fine now

HofiOne commented:
@kira-syslogng retest this please;

HofiOne commented:
@bazsi Could you please take a look at this one more (and hopefully the last time) again?
Thanks!
